src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/MethodCodeNode.cs

```diff
@@ -30,6 +30,22 @@ public MethodCodeNode(MethodDesc method)
             _method = method;
         }
 
+        /// <summary>
+        /// Validates that it will be possible 
+        /// </summary>
+        /// <param name="method"></param>
+        public static void CheckCanGenerateMethod(MethodDesc method)
+        {
+            MethodSignature signature = method.Signature;
+
+            EETypeNode.CheckCanGenerateEEType(signature.ReturnType);
```

Comment:
I whittled this down to the bare minimum the JIT needs, which is it calls getClassAttrib for each method argument and currently the only type loading implication is a check of `ContainsGCPointers`.